AutoRest/Modelers/Swagger/ObjectBuilder.cs

```diff
@@ -45,14 +46,24 @@ public virtual IType BuildServiceType(string serviceTypeName)
             {
                 var enumType = new EnumType();
                 SwaggerObject.Enum.ForEach(v => enumType.Values.Add(new EnumValue { Name = v, SerializedName = v }));
-                if (SwaggerObject.Extensions.ContainsKey("x-ms-enum"))
+                if (SwaggerObject.Extensions.ContainsKey(CodeGenerator.EnumObject))
                 {
-                    enumType.IsExpandable = false;
-                    enumType.Name = SwaggerObject.Extensions["x-ms-enum"] as string;
+                    var enumObject = SwaggerObject.Extensions[CodeGenerator.EnumObject] as Newtonsoft.Json.Linq.JContainer;
+                    if (enumObject != null)
+                    {
+                        enumType.Name= enumObject["name"].ToString();
+                        if (enumObject["expandable"] != null)
+                        {
+                            enumType.IsExpandable = bool.Parse(enumObject["expandable"].ToString());
```

Comment:
If I'm reading this right, `IsExpandable` will be `false` by default. This is a breaking change for everyone currently using `x-ms-enum` that expects it to generate enums rather than string constants. The default should be the current behavior.